logictest: prevent an error with distsql workmem randomization

yuzefovich opened the pull request:
One more query could now hit an error when spilling tuples to disk when distsql workmem is randomized - prevent randomization just for this query.

Fixes: #125607.
Fixes: #125640.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/125609)
<!-- Reviewable:end -->

yuzefovich committed:
logictest: prevent an error with distsql workmem randomization

One more query could now hit an error when spilling tuples to disk when
distsql workmem is randomized - prevent randomization just for this
query.

Release note: None

michae2 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Nice fix!

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125609#-:-O-IfeAM8JhfHtgre3Xm:b-46lt7k)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
